### PR TITLE
chore: Add noir-wasm publishing as a workflow

### PR DESCRIPTION
**Pull request by kevaundray**
# Description

noir wasm is currently being published from a separate repo. This brings publishing of it in here, so its like the other repos

## Problem\*

Resolves 

## Documentation

- [ ] This PR requires documentation updates when merged.

  

  - [ ] I will submit a noir-lang/docs PR.

  

  - [ ] I will request for and support Dev Rel's help in documenting this PR.

  
  

## Additional Context

# PR Checklist\*

- [ ] I have tested the changes locally.
- [ ] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

**Comment by kevaundray** [comment]
Initally just copied the yml file over, need to clean it up

File: .github/workflows/release-noir-wasm.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+      - uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cargo/bin/
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+            target/
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
```
Comment (TomAFrench) [inline .github/workflows/release-noir-wasm.yml:22]: This is incorrect. Will fix in a bit.

File: .github/workflows/release-noir-wasm.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+      - uses: cachix/install-nix-action@v20
+        with:
+          nix_path: nixpkgs=channel:nixpkgs-22.05
```
Comment (kobyhallx) [inline .github/workflows/release-noir-wasm.yml:26]: Let's use updated packages

```suggestion
      - uses: cachix/install-nix-action@v22
        with:
          nix_path: nixpkgs=channel:nixos-23.05
```

**Review by kobyhallx — APPROVED** [review]
LGTM

File: .github/workflows/release-noir-wasm.yml (hunk 1 above)
Comment (TomAFrench) [reply .github/workflows/release-noir-wasm.yml:22]: Actually, let's not bother as caching with nix isn't set up properly anyway.

**Pull request merged**